code/game/machinery/computer/arcade.dm

```diff
@@ -1090,7 +1105,9 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
 	gameStatus = ORION_STATUS_START
 	say("Congratulations, you made it to Orion!")
 	if(obj_flags & EMAGGED)
-		new /obj/item/orion_ship(loc)
+		prizeselect = /obj/item/orion_ship
+		prizevend(user)
+		prizeselect = /obj/item/coin/arcade_token
```

Comment:
what about the other emagged arcade machine prize from the other kind of emagged arcade machine?

Reply:
The Arcade Battle game checks if you win during every action, and in doing so, it dispenses the bomb and the rare sombrero in an emag check. So it doesn't use the prizevend proc to give out it's emagged prize